src/coreComponents/dataRepository/BufferOpsDevice.cpp

```diff
@@ -92,10 +92,12 @@ PackDataDevice( buffer_unit_type * & buffer,
   if( DO_PACKING )
   {
     parallelDeviceStream stream;
-    events.emplace_back( forAll< parallelDeviceAsyncPolicy<> >( stream, var.size(), [=] GEOS_DEVICE ( localIndex ii )
+//    events.emplace_back( forAll< parallelDeviceAsyncPolicy<> >( stream, var.size(), [=] GEOS_DEVICE ( localIndex ii )
+    forAll< parallelDevicePolicy<> >( var.size(), [=] GEOS_DEVICE ( localIndex ii )
     {
       reinterpret_cast< std::remove_const_t< T > * >( buffer )[ ii ] = var.data()[ ii ];
-    } ) );
+    } );
+    //);
```

Comment:
@wrtobin Why is this not something like:
```c++
    std::remove_const_t <T> * const lhs = reinterpret_cast< std::remove_const_t< T > * >( buffer );
   forAll< parallelDevicePolicy<> >( var.size(), [=] GEOS_DEVICE ( localIndex ii )
     {
       lhs[ ii ] = var[ ii ];
     } );
     ```

Reply:
Thats a fine refactor, except we do need `var.data()[ ii ]` since we're packing arbitrary multidimensional slices and just want the underlying device buffer, which inside the loop is on device and outside the loop is on host.

Since the `lhs` is a pinned buffer moving the cast outside the loop should be fine and might be more clear, but the performance should be the same since the cast is a compile-time operation on memory location/alignment more than anything else.